### PR TITLE
feat(crash-feedback): 🚀  add sentry implementation

### PR DESCRIPTION
**Pull request by Maren-Osnabrug**
This PR is related to [#19](https://github.com/storyofams/next-boilerplate/issues/19)

@MrBamBam You had some suggestions to use the sentry/react version and to potentially include an errorboundary(hook) ?

### CODE REVIEW

**Comment by MrBamBam** [comment]
When I heard about using sentry I thought it would be nice to implement it with something that empowers the developers to get more insightfull logging. This to me feels helpfull as it's more than what we currently have, however I think using this in combination with something like ErrorBoundary feels like the right implementation to me.

So tldr; Catch all for unexpected (and unaccounted for errors), but also enable specific logging in the way of using errorBoundaries to handle errors gracefully (while still getting logged).

**Comment by BJvdA** [comment]
It would be very useful to also upload the sourcemap to sentry at build time, which helps in debugging errors. Can you add this?
Also would be nice if you could add a bit of documentation on this, as it requires setting up some env vars, generating an api key in sentry, etc. :)

Regarding the error boundary, to me it would be fine if it was a different issue / branch too, since it's not required for sentry to function.

File: next.config.js [hunk 1]
```diff
@@ -1,17 +1,121 @@
...
+          stripPrefix: ['webpack://_N_E/'],
+          urlPrefix: `~${basePath}/_next`,
+          release: CI_COMMIT_SHA,
+        }),
+      );
+    }
+
     return config;
   },
+  target: 'serverless',
```
Comment (BJvdA) [inline next.config.js:93]: I dont think the `target`, `transformManifest`, and `workboxOpts` should be in this feat, not related to sentry

File: env.example [hunk 2]
```diff
@@ -0,0 +1,12 @@
+NODE_ENV=
+PRODUCTION=
+
+SENTRY_ORG=
+SENTRY_PROJECT=
+SENTRY_AUTH_TOKEN=
+NEXT_PUBLIC_SENTRY_DSN=
+CI_COMMIT_SHA=
+
+NEXT_IS_SERVER=
```
Comment (BJvdA) [inline env.example:10]: This should not be defined in .env, but come from CICD

File: env.example [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+NEXT_PUBLIC_COMMIT_SHA=
```
Comment (BJvdA) [inline env.example:11]: This should not be defined in .env, but come from CICD

File: env.example [hunk 4]
```diff
@@ -0,0 +1,12 @@
+NODE_ENV=
+PRODUCTION=
+
+SENTRY_ORG=
+SENTRY_PROJECT=
+SENTRY_AUTH_TOKEN=
+NEXT_PUBLIC_SENTRY_DSN=
+CI_COMMIT_SHA=
```
Comment (BJvdA) [inline env.example:8]: This should not be defined in .env, but come from CICD

File: next.config.js (hunk 1 above)
Comment (Maren-Osnabrug) [reply next.config.js:93]: I can remove them? it came from merging in master to resolve conflicts, but then the branch was retargetted

**Review by BJvdA — DISMISSED** [review]

**Review by MrBamBam — APPROVED** [review]

**Comment by MrBamBam** [comment]
closes issue #19

**Comment by MrBamBam** [comment]
closes #19

**Pull request merged**